frontends/web/src/components/confirm/Confirm.tsx

```diff
@@ -14,85 +14,71 @@
  * limitations under the License.
  */
 
-import { Component} from 'react';
-import { translate, TranslateProps } from '../../decorators/translate';
+import { FunctionComponent, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import { SimpleMarkup } from '../../utils/markup';
 import { Dialog, DialogButtons } from '../dialog/dialog';
 import { Button } from '../forms';
 
-let confirmation: (message: string, callback: (response: boolean) => void, customButtonText?: string) => void;
+/**
+ * shows an alert when called, triggers the callback with user reply  
+ */
+export let confirmation: (message: string, callback: (response: boolean) => void, customButtonText?: string) => void;
 
 interface State {
     active: boolean;
+    callback?: (input: boolean) => void;
     message?: string;
     customButtonText?: string;
 }
 
-class Confirm extends Component<TranslateProps, State> {
-    private callback!: (input: boolean) => void; // Set within confirmation
-
-    constructor(props: TranslateProps) {
-        super(props);
-        confirmation = this.confirmation;
-        this.state = {
-            active: false,
-        };
-    }
+/**
+ * Confirm alert that activates on confirmation module export call,
+ * this should be mounted only once in the App
+ */
+export const Confirm: FunctionComponent = () => {
+    const [state, setState] = useState<State>({ active: false })
+    const { t } = useTranslation();
 
-    private confirmation = (message: string, callback: (response: boolean) => void, customButtonText?: string) => {
-        this.callback = callback;
-        this.setState({
+    confirmation = (message: string, callback: (response: boolean) => void, customButtonText?: string) => {
+        setState({
             active: true,
+            callback,
             message,
             customButtonText,
         });
     }
 
-    private respond = (input: boolean) => {
-        this.callback(input);
-        this.setState({
+    const respond = (response: boolean) => () => {
+        state.callback!(response);
+        setState({
             active: false,
         });
     }
```

Comment:
Do you prefer curry over just `onClick={() => respond(false)}`?

(I think curring is ok, but I had to check the code twice)